### ISSUE TITLE
geth 1.13.11 cannot sync with prysm 4.2.0 or above 4.2.0

### ISSUE THREAD

**Issue by tunlong** [prysmaticlabs/prysm#13557]
### Describe the bug

At first, everything was normal, but it couldn't reboot. Once restarted and resynchronized, geth couldn't catch up. I tried the RC version of the beacon, but it didn't work either. Finally, I rolled back to version 4.1.1, and everything was fine again.

beacon shows some errors

<img width="1113" alt="image" src="https://github.com/prysmaticlabs/prysm/assets/32350461/d425f452-a6f4-485f-911b-326a298668eb">

geth stucks at age 12m. Although geth gets stuck, the geth console command (eth.syncing) shows that synchronization is completed. 

<img width="1461" alt="image" src="https://github.com/prysmaticlabs/prysm/assets/32350461/3780ebf3-8717-4b28-8d5e-0388e969ad9e">

### Has this worked before in a previous version?

```text
beacon 4.1.1 is working fine.
```

### 🔬 Minimal Reproduction

1.stop beacon and geth
2.start to sync again
3.geth stucks

### Error

```text
[2024-01-30 20:05:03] ERROR blockchain: Could not process slots to get payload attribute error=could not process slots: context deadline exceeded
```

### Platform(s)

Linux (x86)

### What version of Prysm are you running? (Which release)

4.2.0

**Comment by prestonvanloon** [comment]
Potentially related
- https://github.com/ethereum/go-ethereum/issues/28534
- https://github.com/ethereum/go-ethereum/issues/28143
- https://github.com/ethereum/go-ethereum/issues/28079

**Comment by tunlong** [comment]
> This seems like a geth issue. In your geth logs, it seems to take hundreds of milliseconds to process a single block, sometimes more than one second. It should be much faster than that. Typical processing is less than 40ms to 150ms.
> 
> When I've seen this in the past, I deleted the geth db and resynced geth. It's possible that there was an improper shutdown and state healing is ongoing in the background? See [ethereum/go-ethereum#28855 (comment)](https://github.com/ethereum/go-ethereum/issues/28855#issuecomment-1905054466)

Thank you. But why did beacon 4.1.1 work fine. I just replaced beacon version to 4.1.1 and didn't do anything else. Before trying 4.1.1, restarting the machine or replacing RC version didn't help.

**Comment by nisdas** [comment]
We have a rc here:
https://github.com/prysmaticlabs/prysm/releases/tag/v4.2.2-rc.0

If this goes well, it will be our next release. You can give it a try to see if it resolves your issue

**Comment by prestonvanloon** [comment]
@keithchew Do you have any error level logs prior to that one? It should have printed at least one log immediately before that to explain why it was unable to prune a directory.

**Comment by prestonvanloon** [comment]
Thanks @keithchew. The unable to prune directory issue is something we are debugging on your log is very helpful. It shouldn't be a problem at runtime and you could ignore it for now.

The workaround is to delete the directory `$DATADIR/blobs/0x9265c01e6d2fdf61df34b0b025b61a19a1040a78ff13748f634610d4342ac82d`

**Comment by prestonvanloon** [comment]
@keithchew Following up on this. We did find another bug where blobs were not being saved properly. The issue you mentioned https://github.com/prysmaticlabs/prysm/issues/13557#issuecomment-1930767104 has been resolved in #13648
